wezterm-gui/src/termwindow/charselect.rs

```diff
@@ -268,45 +269,57 @@ fn compute_matches(selection: &str, aliases: &[Alias], group: CharSelectGroup) -
         } else {
             None
         };
+        // Use a hash map to remove duplicates, as the same emoji may match multiple times with a different name.
+        let mut matches = HashMap::<char, MatchResult>::new();
 
         let start = std::time::Instant::now();
-        let mut scores: Vec<MatchResult> = aliases
-            .iter()
-            .enumerate()
-            .filter_map(|(row_idx, entry)| {
-                let alias_result = matcher
-                    .fuzzy_match(&entry.name, selection)
-                    .map(|score| MatchResult::new(row_idx, score, selection, aliases));
-                match &numeric_selection {
-                    Some(sel) => {
-                        let codepoints = entry.codepoints();
-                        if codepoints == *sel {
-                            Some(MatchResult {
+        for (row_idx, entry) in aliases.iter().enumerate() {
+            let character = match entry.character {
+                Character::Unicode { value: c, .. } => c,
+                Character::Emoji(d) => d.as_str().chars().next().unwrap(),
```

Comment:
The unwrap here caught my eye; I try to avoid unqualified unwraps and prefer returning errors or using `.expect` to annotate the reason why a panic is acceptable.

However, I think this logic is flawed: Emoji can be comprised of multiple codepoints so hashing based on the first codepoint will lead to falsely skipping certain results; this will be most prevalent for sequences that vary by skin tone, or that combine gender and vocation to produce alternative versions of certain emoji.

What I think is more appropriate here is taking the textual representation of the entry and hashing that.

In other words, use the result of calling `entry.glyph()` as the key to the hash table.